test.js

```diff
@@ -290,6 +290,35 @@ function runTests(options) {
         });
       }));
     });
+    it('should watch removed and re-added directories', function(done) {
+      var unlinkSpy = sinon.spy(function unlinkSpy(){});
+      var addSpy = sinon.spy(function addSpy(){});
+      var parentPath = getFixturePath('subdir');
+      var subPath = getFixturePath('subdir/dir');
+      watcher
+        .on('unlinkDir', unlinkSpy)
+        .on('addDir', addSpy)
+        .on('ready', d(function() {
+          fs.mkdirSync(parentPath);
+          d(function() {
+            fs.rmdirSync(parentPath);
+            d(function() {
+              fs.mkdirSync(parentPath);
+              d(function() {
+                fs.mkdirSync(subPath);
+                waitFor([unlinkSpy], d(function() {
```

Comment:
This assertion should move up to happen right after the directory is removed, and the recreation to happen within that callback after this assertion has passed.